langchain/src/vectorstores/vectara.ts

```diff
@@ -41,30 +49,177 @@ export interface VectaraFile {
+export interface VectaraFilter extends BaseCallbackConfig {
+  // The start position in the result set
+  start?: number;
   // Example of a vectara filter string can be: "doc.rating > 3.0 and part.lang = 'deu'"
   // See https://docs.vectara.com/docs/search-apis/sql/filter-overview for more details.
   filter?: string;
-  // Improve retrieval accuracy by adjusting the balance (from 0 to 1), known as lambda,
+  // Improve retrieval accuracy using Hybrid search, by adjusting the value of lambda (0...1)
   // between neural search and keyword-based search factors. Values between 0.01 and 0.2 tend to work well.
   // see https://docs.vectara.com/docs/api-reference/search-apis/lexical-matching for more details.
   lambda?: number;
-  // The number of sentences before/after the matching segment to add to the context.
+  // See Vectara Search API docs for more details on the following options: https://docs.vectara.com/docs/api-reference/search-apis/search
   contextConfig?: VectaraContextConfig;
+  mmrConfig?: MMRConfig;
 }
 
-/**
- * Interface for the context configuration used in Vectara API calls.
- */
-export interface VectaraContextConfig {
-  // The number of sentences before the matching segment to add. Default is 2.
-  sentencesBefore?: number;
-  // The number of sentences after the matching segment to add. Default is 2.
-  sentencesAfter?: number;
+const DEFAULT_FILTER: VectaraFilter = {
+  start: 0,
+  filter: "",
+  lambda: 0.0,
+  contextConfig: {
+    sentencesBefore: 2,
+    sentencesAfter: 2,
+    startTag: "<b>",
+    endTag: "</b>",
+  },
+  mmrConfig: {
+    enabled: false,
+    mmrTopK: 0,
+    diversityBias: 0.0,
+  },
+};
+
+interface SummaryResult {
+  documents: Document[];
+  scores: number[];
+  summary: string;
+}
+
+export interface VectaraRetrieverInput {
+  vectara: VectaraStore;
+  topK: number;
+  summaryConfig?: VectaraSummary;
+  callbacks?: Callbacks;
+  tags?: string[];
+  metadata?: Record<string, unknown>;
+  verbose?: boolean;
+}
+
+export class VectaraRetriever extends Runnable<string, Document[]> {
+  static lc_name() {
+    return "VectaraRetriever";
+  }
+
+  lc_namespace = ["langchain", "retrievers", "vectaraRetriever"];
+
+  callbacks?: Callbacks;
+
+  tags?: string[];
+
+  metadata?: Record<string, unknown>;
+
+  verbose?: boolean;
+
+  private vectara: VectaraStore;
+  
+  private topK: number;
+
+  private summaryConfig: VectaraSummary;
+
+  constructor(fields: VectaraRetrieverInput) {
+    super(fields);
+    this.callbacks = fields?.callbacks;
+    this.tags = fields?.tags ?? [];
+    this.metadata = fields?.metadata ?? {};
+    this.verbose = fields?.verbose ?? false;
+    this.vectara = fields.vectara;
+    this.topK = fields.topK ?? 10;
+    this.summaryConfig = fields.summaryConfig ?? { enabled: false, maxSummarizedResults: 0, responseLang: "eng" };
+  }
+
+  async invoke(input: string, options?: RunnableConfig): Promise<Document[]> {
+    return this.getRelevantDocuments(input, options);
```

Comment:
This is already present in the base class, no need to override:

https://github.com/langchain-ai/langchainjs/blob/main/langchain-core/src/retrievers.ts#L56

Reply:
But notice VectaraRetriever is does not extend BaseRetriever. won't it be necessary in this case?

Reply:
Why not just pass `summary` as a constructor argument and extend `BaseRetriever`?

Reply:
yes that should work. Thanks!